### PR TITLE
hostap: add pwe config for station

### PR DESCRIPTION
**Pull request by Rex-Chen-NXP**
Add pwe config for station connect cmd.

### CODE REVIEW

File: include/zephyr/net/wifi_mgmt.h [hunk 1]
```diff
@@ -402,6 +402,10 @@ struct wifi_connect_req_params {
 	uint8_t bssid[WIFI_MAC_ADDR_LEN];
 	/** Connect timeout in seconds, SYS_FOREVER_MS for no timeout */
 	int timeout;
+	/** PWE derivation */
+	int sae_pwe;
```
Comment (krish2718) [inline include/zephyr/net/wifi_mgmt.h:406]: As this is only for WPA3, IMHO we should extend the https://github.com/zephyrproject-rtos/zephyr/blob/4bd9522e1941073d8a942a36d526c27b158e91de/include/zephyr/net/wifi.h#L40 security type to include the variants

https://w1.fi/cgit/hostap/tree/wpa_supplicant/wpa_supplicant.conf#n463 

* `WIFI_SECURITY_TYPE_SAE` = `WIFI_SECURITY_TYPE_SAE_HNP` 
* `WIFI_SECURITY_TYPE_SAE_H2E` 
* `WIFI_SECURITY_TYPE_SAE_AUTO` 

WDYT?

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: include/zephyr/net/wifi_mgmt.h [hunk 2]
```diff
@@ -402,6 +402,10 @@ struct wifi_connect_req_params {
...
+	/** PWE configed or not */
+	bool pwe_configed;
```
[the remark this answers is not in the document]
Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi_mgmt.h]: Updated as comment.

File: include/zephyr/net/wifi_mgmt.h (hunk 1 above)
Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi_mgmt.h:406]: No plan to add WIFI_SECURITY_TYPE_SAE_HNP, WIFI_SECURITY_TYPE_SAE_H2E, WIFI_SECURITY_TYPE_SAE_AUTO in wifi_security_type, we added WIFI_SECURITY_TYPE_SAE and sae_pwe type o support all above SAE types.

Comment (krish2718) [reply include/zephyr/net/wifi_mgmt.h:406]: But isn't it simpler to extend security types than adding another parameters specifically for SAE? We should separate API vs implementation, i.e., don't need to expose WPA supplicant internals 121 to the Wi-Fi management, we can abstract it out and then let `supp_api.c` translate to WPA supplicant calls.

Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi_mgmt.h]: Updated as comments, pls help to review again, thanks!

File: include/zephyr/net/wifi.h [hunk 3]
```diff
@@ -46,8 +46,12 @@ enum wifi_security_type {
 	WIFI_SECURITY_TYPE_PSK,
 	/** WPA2-PSK-SHA256 security. */
 	WIFI_SECURITY_TYPE_PSK_SHA256,
-	/** WPA3-SAE security. */
-	WIFI_SECURITY_TYPE_SAE,
+	/** WPA3-SAE security with hunting-and-pecking loop. */
+	WIFI_SECURITY_TYPE_SAE_HNP,
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:50]: To make it backwards compatible, you can do below
```suggestion
	WIFI_SECURITY_TYPE_SAE_HNP = WIFI_SECURITY_TYPE_SAE,
```

Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi.h]: Updated

**Review by krish2718 — APPROVED** [review]
Please fix the formatting issues.

**Comment by Rex-Chen-NXP** [comment]
Hi @krish2718 @jukkar 
Code formatted, could you help to merge pr?

**Comment by krish2718** [comment]
> Hi @krish2718 @jukkar Code formatted, could you help to merge pr?

image
Still showing formatting issues.

**Comment by Rex-Chen-NXP** [comment]
> > Hi @krish2718 @jukkar Code formatted, could you help to merge pr?
> 
> image Still showing formatting issues.

Should I also fixed the merged code format, right?

**Comment by krish2718** [comment]
Yeah, please do as a separate commit for existing code.

**Review by jukkar — APPROVED** [review]

File: modules/hostap/src/supp_api.c [hunk 4]
```diff
@@ -412,6 +414,15 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 				}
 			}
 
+			if (params->security == WIFI_SECURITY_TYPE_SAE_H2E ||
+			    params->security == WIFI_SECURITY_TYPE_SAE_AUTO) {
+				params->sae_pwe =
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:419]: This breaks the build, there is no `sae_pwe` in the params structure, this should be a local variable. Sorry, missed this in the review.

Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c:419]: Should I raise a urge pr to fix it?

Comment (krish2718) [reply modules/hostap/src/supp_api.c:419]: Already did https://github.com/zephyrproject-rtos/zephyr/pull/77274

**Pull request merged**